src/server/saved_objects/service/lib/search_dsl/query_params.test.js

```diff
@@ -50,61 +50,189 @@ const MAPPINGS = {
+        ]
+      }
+    };
+  }
+
+  return {
+    bool: {
+      must: [{ term: { type } }],
+      must_not: [{ exists: { field: 'namespace' } }]
+    }
+  };
+};
+
 describe('searchDsl/queryParams', () => {
-  describe('{}', () => {
-    it('searches for everything', () => {
-      expect(getQueryParams(MAPPINGS))
-        .toEqual({});
+  describe('no parameters', () => {
+    it('searches for all known types without a namespace specified', () => {
+      expect(getQueryParams(MAPPINGS, SCHEMA))
+        .toEqual({
+          query: {
+            bool: {
+              filter: [{
+                bool: {
+                  should: [
+                    createTypeClause('pending'),
+                    createTypeClause('saved'),
+                    createTypeClause('global'),
+                  ],
+                  minimum_should_match: 1
+                }
+              }]
+            }
+          }
+        });
     });
   });
 
-  describe('{type}', () => {
-    it('includes just a terms filter', () => {
-      expect(getQueryParams(MAPPINGS, 'saved'))
+  describe('namespace', () => {
+    it('filters namespaced types for namespace, and ensures namespace agnostic types have no namespace', () => {
+      expect(getQueryParams(MAPPINGS, SCHEMA, 'foo-namespace'))
         .toEqual({
           query: {
             bool: {
-              filter: [
-                {
-                  term: { type: 'saved' }
+              filter: [{
+                bool: {
+                  should: [
+                    createTypeClause('pending', 'foo-namespace'),
+                    createTypeClause('saved', 'foo-namespace'),
+                    createTypeClause('global'),
+                  ],
+                  minimum_should_match: 1
                 }
-              ]
+              }]
             }
           }
         });
     });
   });
 
-  describe('{type,filters}', () => {
-    it('includes filters and a term filter for type', () => {
-      expect(getQueryParams(MAPPINGS, 'saved', null, null, [{ terms: { foo: ['bar', 'baz'] } }]))
+  describe('type (singular, namespaced)', () => {
+    it('includes a terms filter for type and namespace not being specified', () => {
+      expect(getQueryParams(MAPPINGS, SCHEMA, null, 'saved'))
         .toEqual({
           query: {
             bool: {
-              filter: [
-                { terms: { foo: ['bar', 'baz'] } },
-                {
-                  term: { type: 'saved' }
+              filter: [{
+                bool: {
+                  should: [
+                    createTypeClause('saved'),
+                  ],
+                  minimum_should_match: 1
                 }
-              ]
+              }]
+            }
+          }
+        });
+    });
+  });
+
+  describe('type (singular, global)', () => {
+    it('includes a terms filter for type and namespace not being specified', () => {
+      expect(getQueryParams(MAPPINGS, SCHEMA, null, 'saved'))
```

Comment:
Did you mean to test the `global` type here instead? This duplicates the test above.

Reply:
Yeah... thanks again.